hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieHadoopFsRelationFactory.scala

```diff
@@ -94,7 +94,15 @@ abstract class HoodieBaseHadoopFsRelationFactory(val sqlContext: SQLContext,
     tableConfig.getPartitionFields.orElse(Array.empty).toSeq
   } else {
     //it's custom keygen
-    CustomAvroKeyGenerator.getTimestampFields(HoodieTableConfig.getPartitionFieldsForKeyGenerator(tableConfig).orElse(java.util.Collections.emptyList[String]())).asScala.toSeq
+    val timestampFieldsOpt = CustomAvroKeyGenerator.getTimestampFields(
+      HoodieTableConfig.getPartitionFieldsForKeyGenerator(tableConfig).orElse(java.util.Collections.emptyList[String]()))
```

Comment:
I think there is a case where they use custom keygen but only use string fields. If it's not too hard we should make it so we don't need to read them

Reply:
Sorry! Did not get this. Can you please elaborate on this case?

Reply:
ok. So they use custom keygen. There are 3 cases:

1. they have some timestamp cols:
```
partition cols are "part1:timestamp,part2:string,part3:timestamp"
 ```
 In this scenario we want 
 ```
 partitionColumnsToRead = [part1, part3]
 ```
 
 2. they have no timestamp cols:
 ```
 partition cols are "part1:string,part2:string,part3:string"
```
in this scenario we want
```
 partitionColumnsToRead = []
```
3. we don't know because it is a 0.x table
```
partition cols are "part1,part2,part3"
```
we must have 
```
 partitionColumnsToRead = [part1, part2, part3]
```
because we don't know which ones are timestamp


To simplify that to 1 sentence: "partitionColumnsToRead needs to be the list of partition columns that could be timestamp cols"

Reply:
@lokeshj1703 I just had an idea. What is the schema of the partition cols? Wouldn't the schema tell us if it is timestamp? Or is it possible to have string type and still be a timestamp col?

Reply:
All three cases are handled currently. I see there is no test added for the API - Added that as well.